data/examples/declaration/signature/minimal/minimal-out.hs

```diff
@@ -0,0 +1,12 @@
+class Foo a where
+  {-# MINIMAL (==) | ((/=), foo) #-}
+  {-# MINIMAL
+    a
+    | ( b, c, d
```

Comment:
I don't understand why `d | e` is not on the same line even though it is on the same line in the input. I'm not willing to spend time figuring this out though.

Reply:
It's because `e, f` is on multiple lines. So by extension the `|` in `b, c, d | e, f` is also on multiple lines.

It's similar to say formatting:

```haskell
[a, b, (c,
 d)]
```

to:

```haskell
[ a
, b
, ( c
  , d
  )
]
```

Or something similar

Reply:
So it associates like this `d | (e, f)`?

Reply:
The `,` (**and**) has a stronger precedence over `|` (**or**), so yes it associates like this: `(b, c, d) | (e, f)`
I just used the AST for the binary function given to me by GHC, I didn't explicitly write any precedence code myself :smile:

Reply:
Yeah, I see that. Just wanted to confirm that my understanding is correct.